Create ddl.sql

trevorcummingsCCS opened the pull request:
Inital db script for tenders-api

trevorcummingsCCS committed:
Create ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Create drop_tables.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Create provision-tenders-api-database.sh

tommyb82 commented on tenders-api/ddl.sql:
```diff
@@ -0,0 +1,36 @@
+CREATE TABLE procurement_projects (project_id                  UUID PRIMARY KEY,
+                                   commercial_agreement_number VARCHAR(20)   NOT NULL,
+                                   lot_number                  VARCHAR(20)   NOT NULL,
+                                   jaggaer_project_id          VARCHAR(2000) NOT NULL,
```
Can probably get away with a smaller column size for `jaggaer_project_id` (usually something like `tender_1234`)

tommyb82 commented on tenders-api/ddl.sql:
```diff
@@ -0,0 +1,36 @@
+                                   updated_at                  TIMESTAMP);
+                                   
+CREATE TABLE procurement_events  (event_id         VARCHAR(100) PRIMARY KEY, -- This comprises of the OCID as the prfix and an a unique id for the procurement
+                                  jaggaer_event_id VARCHAR(2000) NOT NULL,
```
Likewise, `jaggaer_event_id` should not be anywhere near this size

tommyb82 reviewed (approved):
Looking good - provisioned on a GPaaS PG instance, couple of minor comments but great 👍

trevorcummingsCCS commented:
Ok will reduce them in size later just wanted to get something there that worked.

trevorcummingsCCS committed:
Update ddl.sql

Updated project id from UUID to serial in table procurement_projects. Changed event id to serial from varchar in table procure_events.

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

changed ocds_prefix to ocid_prefix

tommyb82 commented on tenders-api/ddl.sql:
```diff
@@ -0,0 +1,38 @@
+V0.0.1       Initial Release
+*/
+
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
```
Can drop this extension now
